### ISSUE TITLE
[HOLD for payment 2023-11-13] [HOLD for payment 2023-11-09] [$500] Chat - Tapping "Mark as unread option" not showing green line above message

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#30684]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
**Issue found when executing PR:** https://github.com/Expensify/App/pull/30485

**Version Number:** 1.3.94-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause-Internal Team
**Slack conversation:** @

## Action Performed:
1. Go to https://staging.new.expensify.com/
2. Tap on a report
3. Send a message
4.Long press the message sent and select "mark as unread"

## Expected Result:
Tapping "Mark as unread option" must show green line above message and "new message" tag on top of chat.

## Actual Result:
Tapping "Mark as unread option" is not showing green line above message and "new message" tag on top of chat.

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/115492554/9cb8a744-e18b-4cd3-9b3e-2e069b13973e

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/115492554/5583394c-950f-4f6b-98c6-fa714f411c5a

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/115492554/df184356-ed54-43f4-acc9-97ddfcc61946

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01ae8b38a2373a909a</li>
        <li>Upwork Job ID: 1719689014318288896</li>
        <li>Last Price Increase: 2023-11-01</li>
    </ul>
</details>

**Comment by erquhart** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Marking a new message unread does not cause the message to be marked unread until page reload.

### What is the root cause of that problem?
Introduced in this PR: https://github.com/Expensify/App/pull/29860

Because the message creation time is always after `userActiveSince.current`, the message is  not considered to be within the visible threshold, event though it actually is.


### What changes do you think we should make in order to solve the problem?

This was a regression from an otherwise intentional bugfix, logic should be added to show the new marker if the current user marks the message as new without regressing the overall fix from the PR linked above.

### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by situchan** [comment]
@roksanaz can you please check this?

**Comment by situchan** [comment]
@MonilBhavsar I think we should revert https://github.com/Expensify/App/pull/29860 to unblock deploy.
And find better solution which also handles self unread cases.

**Comment by erquhart** [comment]
@situchan just a contributor's two cents - this bug alone, assuming there aren't others, is far preferable to the behavior that #29860 fixes. I personally wouldn't roll it back for this.

**Comment by MonilBhavsar** [comment]
@situchan I reverted, if you could please review https://github.com/Expensify/App/pull/30712. thanks!

**Comment by Beamanator** [comment]
This seems to be fixed in staging, tested on iOS & web 👍

**Comment by jjcoffee** [comment]
No payments due here!

**Comment by stephanieelliott** [comment]
Hey @Li357 can you complete the [BZ checklist](https://github.com/Expensify/App/issues/30684#issuecomment-1796786782) please?

**Comment by Li357** [comment]
Working on it now!

**Comment by Li357** [comment]
Checked off things because it looks like @MonilBhavsar has handled getting another PR out

**Comment by stephanieelliott** [comment]
Summarizing payment on this issue: 

 - Issue reporter: N/A, reported by Applause 
 - Contributor: N/A, internal 
 - Contributor+: @jjcoffee, $500 - extended offer on Upwork, please accept when you get a chance! 
 
 Upwork job is [here](https://www.upwork.com/jobs/~01ae8b38a2373a909a), no bonus on final payout

**Comment by jjcoffee** [comment]
@stephanieelliott No payment is due here to me. I think this was a regression too so @situchan is also not due payment as far as I can tell (feel free to correct me if I'm wrong!).

**Comment by situchan** [comment]
No action needed here. Let's close this!

**Comment by stephanieelliott** [comment]
Cool, thanks guys!
